railties/CHANGELOG.md

```diff
@@ -1,3 +1,47 @@
+*   Credentials commands (e.g. `bin/rails credentials:edit`) now respect
+    `config.credentials.content_path` and `config.credentials.key_path` when set
+    in `config/application.rb`.
+
+    Before:
+
+      * `bin/rails credentials:edit` ignored `RAILS_ENV`, and would always edit
+        `config/credentials.yml.enc`.
+
+      * `bin/rails credentials:edit --environment foo` would create and edit
+        `config/credentials/foo.yml.enc`.
+
+      * If `config.credentials.content_path` or `config.credentials.key_path`
+        was set, `bin/rails credentials:edit` could not be used.  Editing
```

Comment:
```suggestion
        was set, `bin/rails credentials:edit` ignored them.  Editing
```

This is probably more correct.

Reply:
Indeed `bin/rails credentials:edit` would ignore them.  But the point I am making here is that `bin/rails credentials:edit` could not be used to edit the credentials.

If you moved `config/credentials.yml.enc` to a different location and pointed `config.credentials.content_path` there, `bin/rails credentials:edit` would, confusingly, create a new `config/credentials.yml.enc` and edit that.

Or, if you moved `config/credentials/production.key` to a different location and pointed `config.credentials.key_path` there, `bin/rails credentials:edit --production` would create a new, wrong `config/credentials/production.key` and then return a "Couldn't decrypt" error.

Ergo, `bin/rails credentials:edit` could not be used.

Reply:
I have tweaked the wording:

```diff
       * If `config.credentials.content_path` or `config.credentials.key_path`
-        was set, `bin/rails credentials:edit` could not be used.  Editing
-        credentials required using `bin/rails encrypted:edit path/to/credentials
-        --key path/to/key`.
+        was set, `bin/rails credentials:edit` could not be used to edit the
+        credentials.  Editing credentials required using `bin/rails
+        encrypted:edit path/to/credentials --key path/to/key`.
```